l1-contracts/contracts/state-transition/MultisigCommitter.sol

```diff
@@ -0,0 +1,91 @@
+import {SignatureCheckerUpgradeable} from "@openzeppelin/contracts-upgradeable-v4/utils/cryptography/SignatureCheckerUpgradeable.sol";
+import {SignersNotSorted, SignerNotAuthorized, SignatureNotValid} from "../common/L1ContractErrors.sol";
+import {IExecutor} from "./chain-interfaces/IExecutor.sol";
+import {ValidatorTimelock} from "./ValidatorTimelock.sol";
+import {IValidatorTimelock} from "./IValidatorTimelock.sol";
+import {IMultisigCommitter} from "./IMultisigCommitter.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @notice Extended ValidatorTimelock with commit function (optionally) locked
+/// behind providing extra signatures from verifiers. 
+/// @dev The purpose of this contract is to require multiple verifiers check each
+/// commit operation before a sequencer can perform it. Verifiers cannot start 
+/// commit operations. Only Committer can call commitBatchesMultisig, but requires 
+/// providing extra signatures argument
+/// @dev Expected to be deployed as a TransparentUpgradeableProxy.
+contract MultisigCommitter is IMultisigCommitter, ValidatorTimelock, EIP712Upgradeable {
+	/// @dev EIP-712 TypeHash for commitBatchesMultisig
+    bytes32 internal constant COMMIT_BATCHES_MULTISIG_TYPEHASH =
+        keccak256("CommitBatchesMultisig(address chainAddress,uint256 processBatchFrom,uint256 processBatchTo,bytes batchData)");
+
+	bytes32 public constant override COMMIT_VERIFIER_ROLE = keccak256("COMMIT_VERIFIER_ROLE");
+
+	mapping(address chainAddress => uint256 signingThreshold) public signingThreshold;
+
+	constructor(address _bridgehubAddr) ValidatorTimelock(_bridgehubAddr) {
+		_disableInitializers();
+	}
+
+	/// @inheritdoc IValidatorTimelock
+	function initialize(address _initialOwner, uint32 _initialExecutionDelay) external override(ValidatorTimelock, IValidatorTimelock) initializer {
+		__ValidatorTimelock_init(_initialOwner, _initialExecutionDelay);
+		__EIP712_init("MultisigCommitter", "1");
+	}
+
+	/// @inheritdoc IValidatorTimelock
+    function commitBatchesSharedBridge(
+		address _chainAddress,
+        uint256 _processBatchFrom,
+        uint256 _processBatchTo,
+        bytes calldata _batchData
+	) public override(ValidatorTimelock, IValidatorTimelock) {
+		require(signingThreshold[_chainAddress] == 0, "Chain requires verifiers signatures for commit");
+		super.commitBatchesSharedBridge(_chainAddress, _processBatchFrom, _processBatchTo, _batchData);
+	}
+
+	/// @inheritdoc IMultisigCommitter
+	function commitBatchesMultisig(
+        address chainAddress,
+        uint256 processBatchFrom,
+        uint256 processBatchTo,
+        bytes calldata batchData,
+		address[] calldata signers,
+		bytes[] calldata signatures
+    ) external onlyRole(chainAddress, COMMITTER_ROLE) {
+        bytes32 digest = _hashTypedDataV4(keccak256(abi.encode(COMMIT_BATCHES_MULTISIG_TYPEHASH, chainAddress, processBatchFrom, processBatchTo, keccak256(batchData))));
```

Comment:
Do we plan to allow committing only one batch? I understand that we dont have to force it inside contracts, but I am curious whether 2FA signers would need to be aware that multiple batches can be sent at the same time, etc